src/app/[locale]/components/CreateTask.tsx

```diff
@@ -0,0 +1,209 @@
+      type: "",
+      createdOn: "",
+      status: "",
+      assignedTo: "",
+      group: "",
+    },
+  });
+
+  useEffect(() => {
+    if (isLoggedIn === null) {
+      router.push(`/${locale}`);
+    }
+    if (isLoggedIn?.roles.includes("admin") === false) {
+      router.push(`/${locale}/task-list`);
+    }
+  }, [isLoggedIn]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setNotify((prev) => !prev);
+    }, 4000);
+  }, [createTaskId]);
+
+  useEffect(() => {
+    dispatch(getUsers());
+  }, []);
+
+  const selectedGroupId = watch("group");
+
+  useEffect(() => {
+    if (selectedGroupId) {
+      const groupUsers = users.filter((user) => user.group === selectedGroupId);
+      setFilteredUsers(groupUsers);
+    } else {
+      setFilteredUsers([]);
+    }
+  }, [selectedGroupId, users]);
+
+  return (
+    <div className="container w-2/4 m-auto">
+      <form
+        onSubmit={handleSubmit(async (data) => {
+          dispatch(createTask(data));
+          setNotify(true);
+        })}
+      >
+        <div className="bg-slate-100 flex flex-col items-center py-4 mt-4">
+          <h2 className="text-xl font-medium text-center">
+            {t("Create_Task")}
+          </h2>
+          <input
+            type="text"
+            {...register("title", {
+              required: "This is required",
+              maxLength: {
+                value: 30,
+                message: "Maximum length is 30 characters",
+              },
+            })}
+            placeholder="Title"
+            className="w-5/6 p-2 my-2 border-zinc-400 border rounded"
+          />
+          <small className="block text-center text-red-950">
+            {errors.title?.message}
+          </small>
+          <textarea
+            {...register("description", {
+              required: "This is required",
+              maxLength: 200,
+            })}
+            placeholder="Description"
+            className="w-5/6 p-2 my-2 border-zinc-400 border rounded"
+          ></textarea>
+          <small className="block text-center text-red-950">
+            {errors.description?.message}
+          </small>
+          <input
+            type="text"
+            {...register("type", {
+              required: "This is required",
+              maxLength: 20,
+            })}
+            placeholder="Type"
+            className="w-5/6 p-2 my-2 border-zinc-400 border rounded"
+          />
+          <small className="block text-center text-red-950">
+            {errors.type?.message}
+          </small>
+          <input
+            type="date"
+            {...register("createdOn", {
+              required: "This is required",
+              maxLength: 30,
+            })}
+            placeholder="Created On"
+            className="w-5/6 p-2 my-2 border-zinc-400 border rounded"
+          />
+          <small className="block text-center text-red-950">
+            {errors.createdOn?.message}
+          </small>
+          <select
```

Comment:
Maybe its a worth while idea to consider making the select component into a reusable component

Reply:
solved